sklearnex/utils/_array_api.py

```diff
@@ -16,47 +16,124 @@
 
 """Tools to support array_api."""
 
+import itertools
+
 import numpy as np
 
 from daal4py.sklearn._utils import sklearn_check_version
-from onedal.utils._array_api import _get_sycl_namespace
+from onedal.utils._array_api import _asarray, _get_sycl_namespace
 
+# TODO:
+# check the version of skl.
 if sklearn_check_version("1.2"):
     from sklearn.utils._array_api import get_namespace as sklearn_get_namespace
+    from sklearn.utils._array_api import _convert_to_numpy as _sklearn_convert_to_numpy
 
+from onedal._device_offload import dpctl_available, dpnp_available
+
+if dpctl_available:
+    import dpctl.tensor as dpt
+
+if dpnp_available:
+    import dpnp
+
+_NUMPY_NAMESPACE_NAMES = {"numpy", "array_api_compat.numpy"}
 
-def get_namespace(*arrays):
-    """Get namespace of arrays.
 
-    Introspect `arrays` arguments and return their common Array API
-    compatible namespace object, if any. NumPy 1.22 and later can
-    construct such containers using the `numpy.array_api` namespace
-    for instance.
+def yield_namespaces(include_numpy_namespaces=True):
+    """Yield supported namespace.
 
-    This function will return the namespace of SYCL-related arrays
-    which define the __sycl_usm_array_interface__ attribute
-    regardless of array_api support, the configuration of
-    array_api_dispatch, or scikit-learn version.
+    This is meant to be used for testing purposes only.
+
+    Parameters
+    ----------
+    include_numpy_namespaces : bool, default=True
+        If True, also yield numpy namespaces.
+
+    Returns
+    -------
+    array_namespace : str
+        The name of the Array API namespace.
+    """
+    for array_namespace in [
+        # The following is used to test the array_api_compat wrapper when
+        # array_api_dispatch is enabled: in particular, the arrays used in the
+        # tests are regular numpy arrays without any "device" attribute.
+        "numpy",
+        # Stricter NumPy-based Array API implementation. The
+        # array_api_strict.Array instances always have a dummy "device" attribute.
+        "array_api_strict",
+        "dpctl.tensor",
```

Comment:
Should be reverted, since stock scikit-learn testing not well designed for having input data being located on single target device.